test/integration/annotation_spec.js

```diff
@@ -13,49 +13,72 @@
  * limitations under the License.
  */
 
+const { closeAll, loadAndWait } = require("./commons.js");
+
 describe("Annotation highlight", () => {
   describe("annotation-highlight.pdf", () => {
     let pages;
 
     beforeAll(async () => {
-      pages = await Promise.all(
-        global.integrationSessions.map(async session => {
-          const page = await session.browser.newPage();
-          await page.goto(
-            `${global.integrationBaseUrl}?file=/test/pdfs/annotation-highlight.pdf`
-          );
-          await page.bringToFront();
-          await page.waitForSelector("[data-annotation-id='19R']", {
-            timeout: 0,
-          });
-          return page;
-        })
+      pages = await loadAndWait(
+        "annotation-highlight.pdf",
+        "[data-annotation-id='19R']"
       );
     });
 
     afterAll(async () => {
-      await Promise.all(
-        pages.map(async page => {
-          await page.close();
-        })
-      );
+      await closeAll(pages);
     });
 
     it("must show a popup on mouseover", async () => {
       await Promise.all(
-        pages.map(async page => {
+        pages.map(async ([name, page]) => {
           let hidden = await page.$eval(
             "[data-annotation-id='21R']",
             el => el.hidden
           );
-          expect(hidden).toEqual(true);
+          expect(hidden).withContext(`In ${name}`).toEqual(true);
```

Comment:
What does `withContext` do, and what is the advantage of using it here?

Reply:
this function will prepend the string in case of error in the output.
So here it allows to get the name of the browser in case of error.